File: app/src/main/java/jp/hazuki/yuzubrowser/webkit/handler/WebSrcImageHandler.java
package jp.hazuki.yuzubrowser.webkit.handler;

import android.os.Handler;
import android.os.Message;

public abstract class WebSrcImageHandler extends Handler {
    @Override
    public void handleMessage(Message msg) {
        final String url = msg.getData().getString("url");
        if (url == null || url.length() == 0) return;
        handleUrl(url);
    }

    public abstract void handleUrl(String url);
}

File: legacy/src/main/java/jp/hazuki/yuzubrowser/legacy/webkit/handler/WebSrcImageHandler.kt
package jp.hazuki.yuzubrowser.legacy.webkit.handler

import android.os.Handler
import android.os.Looper
import android.os.Message

abstract class WebSrcImageHandler : Handler(Looper.getMainLooper()) {
    override fun handleMessage(msg: Message) {
        val url = msg.data.getString("url")
        if (url.isNullOrEmpty()) return
        handleUrl(url)
    }

    abstract fun handleUrl(url: String)
}
